### PR TITLE
Update gosecret

### PR DESCRIPTION
**Pull request by calvn**
This version will support gosecret 0.5.0.

### CODE REVIEW

**Comment by calvn** [comment]
@maclennann @jdaviscooke @ryanbreen can you guys check if I missed anything?

**Comment by calvn** [comment]
Tests will be coming out on 0.3.1. I wanted to cut this version for release ASAP before switching context back to consul.

**Comment by ryanbreen** [comment]
I'm on paternity, you maniac! 😜

**Comment by calvn** [comment]
I assumed you would be on no matter what. So far your git commits confirm my assumption.

**Comment by ryanbreen** [comment]
My github commit streak was my first child.

**Comment by calvn** [comment]
Adding @swivelhinges for review

File: lib/hiera/backend/gosecret_backend.rb [hunk 1]
```diff
@@ -53,13 +53,24 @@ def decrypt_array(array)
...
+      def decrypt(value)
+        if value != nil and value.is_a?(String) and ( /\[(gosecret(\|[^\]\|]*){4})\]/.match(value) or /{{ *goDecrypt .*}}/.match(value) )
           Hiera.debug("Decrypting gosecret encrypted value: #{value}")
-          args = "-mode=decrypt -keystore=\"#{Config[:gosecret][:keydir]}\" -value=\"#{value}\""
+          args = "-mode=decrypt -keystore=\"#{Config[:gosecret][:keydir]}\" -value='#{value}'"
           return `gosecret #{args}`
         end
       end
+
+      def which(cmd)
```
Comment (maclennann) [inline lib/hiera/backend/gosecret_backend.rb]: Is this used anywhere?

**Comment by maclennann** [comment]
Having never seen hiera, hiera-gosecret, or gosecret code before, I can confidently say that this looks fine to me.

File: lib/hiera/backend/gosecret_backend.rb (hunk 1 above)
Comment (calvn) [inline lib/hiera/backend/gosecret_backend.rb]: It is not, good catch.

**Pull request merged**